### PR TITLE
Convert OrganismDbi.Rnw to OrganismDbi.Rmd

### PR DESCRIPTION
**Pull request by alatiku**
@Bioconductor/sweave2rmd please review my work. I have converted the OrganismDbi.Rnw sweave vignette to OrganismDbi.Rmd file. 

Find attached the PDF [OrganismDbi.pdf](https://github.com/Bioconductor/OrganismDbi/files/11104168/OrganismDbi.pdf) and the HTML file [OrganismDbi.zip](https://github.com/Bioconductor/OrganismDbi/files/11104186/OrganismDbi.zip).

### CODE REVIEW

**Comment by alatiku** [comment]
@jwokaty please review

File: vignettes/OrganismDbi.Rmd
Comment (mcarlsn) [inline vignettes/OrganismDbi.Rmd]: Looks like there should be a carriage line here. According to the original vignette PDF, the format of the passage is: 

are `columns`, `keys`, `keytypes` and `select`. 

So to learn which kinds of data can be retrieved from a package like this we would simply load the package and then call the `columns` method.

File: vignettes/OrganismDbi.Rmd
Comment (mcarlsn) [inline vignettes/OrganismDbi.Rmd]: In the original PDF, "keytype" here is not italicized so this can be changed to regular font.

File: vignettes/OrganismDbi.Rmd
Comment (mcarlsn) [inline vignettes/OrganismDbi.Rmd]: To match the original vignette, we can edit this slightly to: 

"And to extract data, we can use the `select` method. The select method depends on the values from the previous three methods to specify what it will extract. Here is an example that will extract, UCSC transcript names, and gene symbols using Entrez Gene IDs as keys."

File: vignettes/OrganismDbi.Rmd
Comment (mcarlsn) [inline vignettes/OrganismDbi.Rmd]: This part is identical to the original vignette but because I'm assuming that the capital "A" in "Addition" was a typo so we can edit to "In addition to..."

File: vignettes/OrganismDbi.Rmd
Comment (mcarlsn) [inline vignettes/OrganismDbi.Rmd]: Edit text formatting to: "So this `data.frame` indicates both which packages are connected to each other..."

File: vignettes/OrganismDbi.Rmd
Comment (mcarlsn) [inline vignettes/OrganismDbi.Rmd]: Edit text formatting to: "`data.frame`. Because the end result will be a package that treats all the data..."

File: vignettes/OrganismDbi.Rmd
Comment (mcarlsn) [inline vignettes/OrganismDbi.Rmd]: YAML can be slightly updated to add numbered sections and TOC. We can change it to: 

output: 
BiocStyle:: html_document: 
numbered_sections: true 
toc: true 
toc_depth: 4

**Comment by mcarlsn** [comment]
Great work @sirjoka. As a FYI for any future work related to vignette conversion, I'm leaving comments below on changes/ edits. Please note that the fact that you do not have time to make edits within the contribution period will _not_ negatively impact your Sweave2Rmd application. Please let me know if you have any questions! 

- [ ] in the `DESCRIPTION` file, `BiocStyle` and `knitr` need to be added to the `Suggests` list
- [ ] in the `DESCRIPITON`, the line `VignetteBuilder: knitr` needs to be added 
- [ ] any added lines use the same spacing and indents as the existing document (see line comments above) 
- [ ] the R Markdown file is representative of the Sweave document and follows best practices (in the script itself, there are some text lines at the top of the script that extend past the vertical line in RStudio that serves as a guide to cap each line at 80 characters. It’s preferable for each line in the script to stay on the left side of that vertical line. This can be fixed by selecting the text in your script and then clicking on Code > Reflow Comment. It will automatically reformat your script and text.) 
- [x] the `.Rmd` file knits to `HTML`
- [x] `R CMD build` runs without errors or timeouts
- [x] the tarball from `R CMD build` contains the `HTML` (check with the following by substituting the package name and vignette name `tar ztf package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the `DESCRIPTION` file; refer to the contribution guide for acceptable
  formats
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the vignette's YAML; refer to the contribution guide for acceptable formats
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, check that the
  `Authors@R` vector includes the maintainer as specified with `role='cre'`.
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
  is completely removed.
- [x] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [x] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request

**Comment by alatiku** [comment]
Hi @mcarlsn I have made the requested changes please have a look.

Thank you

File: vignettes/OrganismDbi.Rmd
Comment (mcarlsn) [inline vignettes/OrganismDbi.Rmd]: Very very small edit: it looks like the word "very" here extends over the vertical line so if we can hit enter and add that to the next line, then everything will be on the left side of that marker. Thank you for updating this!

**Review by mcarlsn — COMMENTED** [review]
@sirjoka thank you so much for making these changes, your conversation looks awesome! I left one tiny edit but everything else is good to go.

**Comment by alatiku** [comment]
> @sirjoka thank you so much for making these changes, your conversation looks awesome! I left one tiny edit but everything else is good to go.

Thank you. Easy to learn with wonderful mentors. I have made the last edit, please have a look. Thank you again.

**Review by mcarlsn — APPROVED** [review]
Amazing, thanks so much for making that edit @sirjoka. This looks great. Will elevate this vignette to the next step.

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: Please fix the formatting for lines 15-20.

File: vignettes/OrganismDbi.Rmd
Comment (jwokaty) [inline vignettes/OrganismDbi.Rmd]: Could you fix the formatting for the 2 paragraphs in line 22-36 in RStudio via `Code > Reflow Comment`?

File: vignettes/OrganismDbi.Rmd
Comment (jwokaty) [inline vignettes/OrganismDbi.Rmd]: Please reformat this paragraph.

File: vignettes/OrganismDbi.Rmd
Comment (jwokaty) [inline vignettes/OrganismDbi.Rmd]: On lines 111,112 and 129, TxDb and OrgDb aren't packages. I believe they're kinds of packages, so let's just mark them with backticks.

File: vignettes/OrganismDbi.Rmd
Comment (jwokaty) [inline vignettes/OrganismDbi.Rmd]: On lines 167 and 185, replace

    `r Biocpkg("makeOrganism")` package

with

    `makeOrganismPackage`
 
This is the name of a function.

**Review by jwokaty — CHANGES_REQUESTED** [review]
@sirjoka Thanks for this PR. It looks good; however, I noticed a few small changes that should be made before it is merged.

Since we're replacing https://github.com/Bioconductor/OrganismDbi/blob/devel/vignettes/databaseTypes.pdf with a .png, can you please also remove this file, providing a short explanation in your commit?

**Comment by alatiku** [comment]
@jwokaty I have made the necessary formatting changes to the DESCRIPTION and OrganismDbi.Rnw, and removed the databaseType.pdf with the appropriate comment in the commit.
Please have a look when free. Thank you

**Comment by jwokaty** [comment]
Thanks for your contribution!

**Pull request merged**
